### PR TITLE
done the chat implementation and ui fixes

### PR DESCRIPTION
**Pull request by razuahmedjoy**

### CODE REVIEW

File: client/src/Components/Chat/Messages/Messages.jsx [hunk 1]
```diff
@@ -11,107 +11,193 @@ import img3 from '../../../assets/ai logos/image3.png'
...
-                                    <div className="flex items-center  gap-2 w-2/3">
+
+                    {chatLoading ? (
+                        <div className="flex items-center justify-center h-full">
+                            <LoadingIcon color='black' />
+                        </div>
+                    ) : (
+                        <>
+                            {(!messages.length || isNewChat) ?
+                                <div className="flex items-center justify-center h-full">Ask somthing to get response</div> :
```
Comment (pankaj3399) [inline client/src/Components/Chat/Messages/Messages.jsx]: Ask something to get response, wrong spelling

File: server/utils/server/config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import dotenv from 'dotenv';
+import path from 'path';
+
+dotenv.config({ path: path.join(process.cwd(), '.env') });
+
+export default {
+    OPENAI_API_KEY: process.env.OPENAI_API_KEY,
```
Comment (pankaj3399) [inline server/utils/server/config.js:7]: There are more env variables like database connection string, please follow same structure for all envs

File: client/src/Components/Chat/Messages/Messages.jsx [hunk 3]
```diff
@@ -11,107 +11,193 @@ import img3 from '../../../assets/ai logos/image3.png'
...
-        id: 3,
-        role: 'user',
-        message: 'Messages from the user',
-
+import { useEffect, useRef, useState } from "react";
+import { cx } from "../../../hooks/helpers";
+import toast from "react-hot-toast";
+import { useCreateChatMutation } from "../../../redux-rtk/features/chat/chatApi";
+import LoadingIcon from "../../Shared/LoadingIcon/LoadingIcon";
+// import parse from 'html-react-parser';
```
Comment (pankaj3399) [inline client/src/Components/Chat/Messages/Messages.jsx]: Remove this if not needed

File: client/src/Components/Chat/Messages/Messages.jsx [hunk 4]
```diff
@@ -11,107 +11,193 @@ import img3 from '../../../assets/ai logos/image3.png'
...
+        };
+
+        setMessages([
+            ...messages,
+            {
+                ...sendData,
+                content: sendData.message
+            }
+        ])
+        // setIsNewChat(false)
```
Comment (pankaj3399) [inline client/src/Components/Chat/Messages/Messages.jsx]: Remove these lines

File: server/app/models/chatSchema.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+const ChatSchema = new Schema({
+    user: {
+        type: Types.ObjectId,
+        ref: "User",
+        required: [true, 'User required!']
+    },
+    title: {
+        type: String,
+        required: [true, 'Title required!']
+    },
```
Comment (pankaj3399) [inline server/app/models/chatSchema.js:12]: Add the type of model also in this schema

**Comment by razuahmedjoy** [comment]
added another pr

**Pull request closed without merging**